Sources/OpenAPIRuntime/Base/Acceptable.swift

```diff
@@ -0,0 +1,177 @@
+        self.init(doubleValue: value)
+    }
+}
+
+extension Array where Element == QualityValue {
+
+    /// Returns a sorted array of quality values, where the highest
+    /// priority items come first.
+    public func sortedByQuality() -> Self {
+        sorted { a, b in
+            a.doubleValue < b.doubleValue
+        }
+    }
+}
+
+extension Array where Element: AcceptableProtocol {
+
+    /// Returns the default values for the acceptable type.
+    public var defaultValues: Self {
+        Element.defaultValues
+    }
+}
+
+/// A wrapper of an individual content type in the accept header.
+public struct AcceptHeaderContentType<T: AcceptableProtocol>: Sendable, Equatable, Hashable {
+
+    /// The value representing the content type.
+    public var contentType: T
+
+    /// The quality value of this content type.
+    ///
+    /// Used to describe the order of priority in a comma-separated
+    /// list of values.
+    ///
+    /// Content types with a higher priority should be preferred by the server
+    /// when deciding which content type to use in the response.
+    ///
+    /// Also called the "q-factor" or "q-value".
+    public var quality: QualityValue
+
+    /// Creates a new content type from the provided parameters.
+    /// - Parameters:
+    ///   - value: The value representing the content type.
+    ///   - quality: The quality of the content type, between 0.0 and 1.0.
+    /// - Precondition: Priority must be in the range 0.0 and 1.0 inclusive.
+    public init(contentType: T, quality: QualityValue = 1.0) {
+        self.quality = quality
+        self.contentType = contentType
+    }
+
+    /// Returns the default set of acceptable content types for this type, in
+    /// the order specified in the OpenAPI document.
+    public static var defaultValues: [Self] {
+        T.defaultValues.map { .init(contentType: $0) }
+    }
+}
+
+extension AcceptHeaderContentType: RawRepresentable {
+    public init?(rawValue: String) {
+        guard let validMimeType = OpenAPIMIMEType(rawValue) else {
+            // Invalid MIME type.
+            return nil
+        }
+        let quality: QualityValue
+        if let rawQuality = validMimeType.parameters["q"] {
+            guard let parsedQuality = QualityValue(rawValue: rawQuality) else {
+                // Invalid quality parameter.
+                return nil
+            }
+            quality = parsedQuality
+        } else {
+            quality = 1.0
+        }
+        guard let typeAndSubtype = T(rawValue: validMimeType.kind.description.lowercased()) else {
+            // Invalid type/subtype.
+            return nil
+        }
+        self.init(contentType: typeAndSubtype, quality: quality)
+    }
+
+    public var rawValue: String {
+        contentType.rawValue + (quality.isDefault ? "" : "; q=\(quality.rawValue)")
+    }
+}
+
+extension AcceptHeaderContentType {
+
+    // TODO: Can we spell this as an extension of an array?
```

Comment:
I think the spelling would be something like:

```swift
extension Array {
  func sortedByQuality<T: AcceptableProtocol>() -> [AcceptHeaderContentType<T>] where Element == AcceptHeaderContentType<T> { 
    // ...
  } 
}


```

Reply:
Ah that's what I was looking for – thanks! 🙏